scripts/jenkins/cloud/ansible/roles/heat-generator/templates/heat-template.yaml

```diff
@@ -71,8 +72,22 @@ resources:
       network_id: { get_resource: {{ name }}_network }
 {%   if network.cidr is defined %}
       cidr: "{{ network.cidr }}"
+{%       if ( network.cidr | ipv6 ) %}
+      ip_version: 6
+{%          if network.is_conf or network.is_mgmt %}
+      ipv6_ra_mode: dhcpv6-stateful
+      ipv6_address_mode: dhcpv6-stateful
+{%          endif %}
+{%       else %}
+      ip_version: 4
+{%       endif %}
 {%   elif network.allocation_pools is not defined %}
       cidr: { get_param: default_subnet_cidr }
+      ip_version: {{ ipv6 | ternary(6, 4) }}
+{%       if network.is_conf or network.is_mgmt %}
+      ipv6_ra_mode: dhcpv6-stateful
+      ipv6_address_mode: dhcpv6-stateful
```

Comment:
It would be nice if we could get rid of this repetition with line 78 and 79, but I can figure out a logical path for it. 
may be check network.is conf at another place? ...

Reply:
Yes, we need something like "we want DHCP" instead of this combination of testing for `is_conf` or `is_mgmt`. I am working on such logic for SOC-9828.

Reply:
You could add some logic in the jinja earlier to set flags/values based on specific combinations of conditions, and then change the code which generates the output to check for those flags and/or use the determined values when generating the output